app/src/main/kotlin/org/stypox/dicio/io/wake/oww/OpenWakeWordDevice.kt

```diff
@@ -32,10 +32,16 @@ class OpenWakeWordDevice(
     private val melFile = FileToDownload(MEL_URL, File(owwFolder, "melspectrogram.tflite"))
     private val embFile = FileToDownload(EMB_URL, File(owwFolder, "embedding.tflite"))
     private val wakeFile = FileToDownload(WAKE_URL, File(owwFolder, "wake.tflite"))
-    private val allModelFiles = listOf(melFile, embFile, wakeFile)
+    private val userWakeFile = userWakeFile(appContext)
+    private val userWakeFileExists = userWakeFile.exists()
+    private val allModelFiles =
+        // wakeFile is not needed if we want to use the userWakeFile instead
+        if (userWakeFileExists) listOf(melFile, embFile)
+        else listOf(melFile, embFile, wakeFile)
 
     private val audio = FloatArray(OwwModel.MEL_INPUT_COUNT)
     private var model: OwwModel? = null
+    private var needsReload: Boolean = false
```

Comment:
Since you've removed the use of this variable, I would recommend removing it too.